lib/mayaUsd/fileio/orphanedNodesManager.cpp

```diff
@@ -327,10 +330,38 @@ void OrphanedNodesManager::recursiveSwitch(
         }
     } else {
         auto childrenComponents = trieNode->childrenComponents();
+        // Figure out the runtime ID of the children of the trie node.
+        // It only changes when the node is a gateway node. So we assume
+        // it will have the same runtime ID by default and only retrieve
+        // the new runtime ID if we cross a gateway node.
+        const Ufe::Rtid runtimeId = ufePath.runTimeId();
+        Ufe::Rtid       childRuntimeId = runtimeId;
+        char            childSep = ufePath.getSegments().back().separator();
+        if (Ufe::SceneSegmentHandler::isGateway(ufePath)) {
+            if (Ufe::SceneItem::Ptr sceneItem = Ufe::Hierarchy::createItem(ufePath)) {
+                if (Ufe::Hierarchy::Ptr itemHier = Ufe::Hierarchy::hierarchy(sceneItem)) {
+                    Ufe::Hierarchy::ChildFilter filter
+                        = { Ufe::ChildFilterFlag("InactivePrims", "InactivePrims", 1) };
+                    for (auto child : itemHier->filteredChildren(filter)) {
+                        childRuntimeId = child->runTimeId();
+                        childSep = child->path().getSegments().back().separator();
+                        break;
+                    }
+                }
+            }
+        }
```

Comment:
Unless I'm missing something, I don't think we need to do this.  We are the maya-usd plugin, and we have said that we would not support 3-segment paths --- pulled nodes are always single-segment Maya nodes, for compatibility with existing Maya code.

Therefore, the second segment will always be USD in this plugin, so we can simply get the USD run-time ID from getUsdRunTimeId() in Global.h, and use the USD path component separator '/'.  We can do this `if (Ufe::SceneSegmentHandler::isGateway(ufePath))`